crashtracker/src/crash_handler.rs

```diff
@@ -173,46 +236,43 @@ fn handle_posix_signal_impl(signum: i32) -> anyhow::Result<()> {
     // In fact, if we look into the code here, we see mallocs.
     // https://doc.rust-lang.org/src/std/backtrace.rs.html#332
     // Do this last, so even if it crashes, we still get the other info.
-    unsafe { emit_backtrace_by_frames(pipe, RESOLVE_FRAMES)? };
+    unsafe {
+        emit_backtrace_by_frames(
+            pipe,
+            config.resolve_frames == CrashtrackerResolveFrames::ExperimentalInProcess,
+        )?
+    };
     writeln!(pipe, "{DD_CRASHTRACK_DONE}")?;
 
     pipe.flush()?;
     // https://doc.rust-lang.org/std/process/struct.Child.html#method.wait
     // The stdin handle to the child process, if any, will be closed before waiting.
     // This helps avoid deadlock: it ensures that the child does not block waiting
     // for input from the parent, while the parent waits for the child to exit.
-    //TODO, use a polling mechanism that could recover from a crashing child
+    // TODO, use a polling mechanism that could recover from a crashing child
     receiver.wait()?;
+    // Calling "free" in a signal handler is dangerous, so don't do that.
     Ok(())
 }
 
-pub fn register_crash_handlers(config: &CrashtrackerConfiguration) -> anyhow::Result<()> {
+// TODO, there is a small race condition here, but we can keep it small
```

Comment:
Minor: Consider explaining the race condition? (It's that the signal handlers aren't setup atomically, right?)